### PR TITLE
LinSrgba, Srgba and Hsla color types

### PR DESCRIPTION
**Pull request by lassade**
The bevy 0.5 made colors operations undefined, this directly affects our ability to animate any color property.

There is a ongoing discussion [here](https://github.com/bevyengine/bevy/discussions/1849) about this.

So this PR does the following:

1. Adds `LinSrgba`, `Srgba`, `Hsla` color types.

Using colors as types allows for predictable behaviour, easer readability, compile time optimizations:

```
// Lerp happens in HSL space but result is stored as sRGB
// a is converted to HSL, b is a HSL, 2 conversions are made (just count the into calls)
let c: sRGB = HSL::lerp(a.into(), b, 0.5).into(); 
```

When conversions from SRGB to HSL and back to SRGB are done often, cache the HSL;

2. `type Color = LinSrgba`

By default the renderer should operate in the linear color space so we define the `Color` type to be in this same space, to avoid any unnecessary conversions;

3. Fixes color constants for both `LinSrgba` and `Srgba`, 

I just used a python script to convert the color constants between each space;

### CODE REVIEW

**Comment by mockersf** [comment]
Before this, I think we should agree on wether to roll our own colors, use [palette](https://crates.io/crates/palette), use [kolor](https://github.com/kabergstrom/kolor) or [rgb](https://crates.io/crates/rgb)

**Comment by lassade** [comment]
I made this PR, because I really want to animate colors and to make easier to test out this approach (which, I think is the best), so nothing is set stone.


I think roiling with our own type is simpler because we can implement any conversions or traits we want;


If you notice I aligned the colors to 16 bytes so they can be ready to use with simd instructions (SSE2), thats because I assumed everything have a alpha, and that is not something I saw in any other crate;


Plus I'm thinking in using the `kolor` crate (once it's ready), because it allows to use our own storage types while having theirs conversions methods;

**Comment by mockersf** [comment]
Sound good to me, I would just like to avoid a breaking change in colors in 0.6 and in 0.7. Breaking twice before 0.6 should be ok.

@kabergstrom when do you feel `kolor` could be ready?

**Comment by kabergstrom** [comment]
I think it's getting there, I added HSV/HSI/HSL, ICtCp and CIELuv this weekend. Did some more work on the docs. What would you like to see in the library?
I need to write the readme, set up CI and add better reference unit tests.

**Comment by lassade** [comment]
I'm blocked by some animator system stuff before I can do some more testing about colors. Do way I think adding using kolor won't break anything, will just add more color types;

File: crates/bevy_render/src/color/conversions.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+// source: https://entropymine.com/imageworsener/srgbformula/
+impl SrgbColorSpace for f32 {
+    fn linear_to_nonlinear_srgb(self) -> f32 {
+        if self <= 0.0 {
+            return self;
+        }
+
+        if self <= 0.0031308 {
```
Comment (NathanSWard) [inline crates/bevy_render/src/color/conversions.rs:15]: For floating-point equality checks, should we be using some sort of `epsilon` or a different crate like `float-cmp`?

**Comment by alice-i-cecile** [comment]
#4648 contains further design direction on color. It's *probably* going to be easier to start from scratch, but I'll leave that decisison to you.

**Comment by nicopap** [comment]
Superseded by #12013

**Pull request closed without merging**